docs/BUILD.md

```diff
@@ -1,6 +1,6 @@
 # Building MsQuic
 
-MsQuic uses [CMake](https://cmake.org/) to generate build files.
+The full MsQuic build system relies on [CMake](https://cmake.org/) (3.16 of better), [.NET Core](https://dotnet.microsoft.com/download/dotnet-core) (3.1 SDK or better) and [Powershell](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows) (7.0 or better) on all platforms.
```

Comment:
It has to be 3.1. because of how the tools work, 5 won't work. Next time we release clog we'll build it for both.